src/opening-period/page/EditOpeningPeriodPage.test.tsx

```diff
@@ -0,0 +1,332 @@
+  override: false,
+  resource: 1,
+  time_span_groups: [
+    {
+      id: 2,
+      period: 1,
+      rules: [],
+      time_spans: [
+        {
+          created: '2020-12-18T10:00:00.000000+02:00',
+          description: {
+            fi: 'Viikonloppuna ovet menevät kiinni tuntia aiemmin',
+            sv: null,
+            en: null,
+          },
+          end_time: '17:00:00',
+          full_day: false,
+          group: 1225,
+          id: 2637,
+          modified: '2020-12-18T10:00:00.000000+02:00',
+          name: {
+            fi: null,
+            sv: null,
+            en: null,
+          },
+          resource_state: ResourceState.SELF_SERVICE,
+          start_time: '12:00:00',
+          weekdays: [6, 7],
+        },
+        {
+          created: '2020-12-18T10:00:00.000000+02:00',
+          description: {
+            fi: 'Arkena ovet menevät kiinni puolta tuntia aiemmin',
+            sv: null,
+            en: null,
+          },
+          end_time: '18:00:00',
+          full_day: false,
+          group: 1225,
+          id: 2636,
+          modified: '2020-12-18T10:00:00.000000+02:00',
+          name: {
+            fi: null,
+            sv: null,
+            en: null,
+          },
+          resource_state: ResourceState.OPEN,
+          start_time: '10:00:00',
+          weekdays: [1, 2, 3, 4, 5],
+        },
+      ],
+    },
+  ],
+};
+
+describe(`<EditNewOpeningPeriodPage />`, () => {
+  beforeEach(() => {
+    jest
+      .spyOn(api, 'getResource')
+      .mockImplementation(() => Promise.resolve(testResource));
+
+    jest
+      .spyOn(api, 'getDatePeriodFormOptions')
+      .mockImplementation(() => Promise.resolve(testDatePeriodOptions));
+
+    jest
+      .spyOn(api, 'getDatePeriod')
+      .mockImplementation(() => Promise.resolve(testDatePeriod));
+  });
+
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it('should render basic date-period data', async () => {
+    let container: HTMLElement;
+
+    await act(async () => {
+      container = render(
+        <EditOpeningPeriodPage resourceId="tprek:8100" datePeriodId="1" />
+      ).container;
+
+      if (!container) {
+        throw new Error(
+          'Something went wrong in rendering of CreateNewOpeningPeriodPage'
+        );
+      }
+    });
+
+    await act(async () => {
+      // Check info data
+      expect(
+        container.querySelector('[data-test="openingPeriodTitle"]')
+      ).toHaveValue(testDatePeriod.name.fi);
+
+      expect(
+        container.querySelector('#openingPeriodOptionalDescription')
+      ).toHaveValue(testDatePeriod.description.fi);
+
+      expect(container.querySelector('#openingPeriodBeginDate')).toHaveValue(
+        '18.12.2020'
+      );
+      expect(container.querySelector('#openingPeriodEndDate')).toHaveValue(
+        '18.01.2021'
+      );
+    });
+  });
+
+  it('should render first the time-span when it has the weekdays in the beginning of the week', async () => {
```

Comment:
Not sure if this could be more clear like this: 'Should render first the time-span which has the weekdays in the beginning of the week'?

Reply:
Fixed